src/olympia/devhub/tests/test_views.py

```diff
@@ -1899,7 +1898,7 @@ def test_header_image(self):
         zip_file = os.path.join(
             settings.ROOT, 'src/olympia/devhub/tests/addons/static_theme.zip'
         )
-        copy_stored_file(zip_file, destination)
+        self.root_storage.copy_stored_file(zip_file, destination)
```

Comment:
I was confused about why sometimes you import `root_storage`, and other times you use `self.root_storage`. I can see that when a test class inherits from `TestCase`, then you can use `self.root_storage`, but why do some of our tests live in classes that inherit from `TestCase` while others do not?

Reply:
No particular reason, it's just the two ways of writing python tests.  Generally if you need a single test for a stand-alone function, a function based test is fine; if there are multiple tests needed, or you need some of the extra setup/functionality available within a `TestClass`, (e.g. you have to mark each function based test as needing database access or it's blocked), then subclass.  Conversely there are a few things that are difficult to do inside classes, like `@pytest.mark.parametrize`

Reply:
Oh, so after re-reading your question it seems maybe you're talking about tests that live in classes that inherit from `TestCase` vs. _tests that live in classes that don't inherit from `TestCase`_.  If so, can you point them out because I don't recall them.